testing/git_cached_repository/git_cached_repository.sh

```diff
@@ -0,0 +1,124 @@
+#!/bin/bash
+
+source ./testing/e2e/init.sh # Must be sourced 1st
+source ./testing/e2e/bazel.sh
+source ./testing/e2e/rules.sh
+source ./testing/e2e/unittest.sh
+source ./testing/e2e/logging.sh
+source ./testing/e2e/utils.sh
+source ./testing/e2e/repos.sh
+source ./testing/e2e/assert.sh
+source ./testing/e2e/testkit.sh
+
+# Must be invoked from the root of the repo.
+ROOT=$PWD
+
+function before_test() {
+  set_up
+  TEST_name=$1
+  TEST_passed="true"
+  create_clean_workspace
+  copy_git_cached_repo_rules_to_workspace
+  set_git_cached_repo_path
+  log_test_name ${TEST_name}
+  # Always cd into the workspace root directory
+  cd ${WORKSPACE_DIR}
+}
+
+function after_test() {
+  tear_down
+  delete_git_cached_directory ${TEST_GIT_CACHE_DIR}
+}
+
+function test_cached_dir_created_as_expected() {
+  before_test "test_cached_dir_created_as_expected"
+
+  # Given I declare a git cache rule with custom cache directory
+  create_git_cached_rule_for_small_repo "014459e6361b66a7b210758d4bf93f3a46ca5e88" ${TEST_GIT_CACHE_DIR}
+
+  # And I create a shell library to reference the git cache rule
+  target_label=$(create_sh_lib_ref_small_repo)
+
+  # And I build the shell library target label
+  bazel build //${target_label} >&${TEST_log} ||
+    echo "Expected git cached directory to get created successfully"
+
+  # Then I expect the repository to be cached successfully
+  expect_folder ${TEST_GIT_CACHE_DIR}
+  after_test
+}
+
+function test_cached_default_dir_created_as_expected() {
+  before_test "test_cached_default_dir_created_as_expected"
+
+  # Given I declare a git cache rule using the default cache directory
+  create_git_cached_rule_for_small_repo "014459e6361b66a7b210758d4bf93f3a46ca5e88"
+
+  # And I create a shell library to reference the git cache rule
+  target_label=$(create_sh_lib_ref_small_repo)
+
+  # And I build the shell library target label
+  bazel build //${target_label} >&${TEST_log} ||
+    echo "Expected git cached default directory to get created successfully"
+
+  # Then I expect the repository to be cached successfully within default cache library
+  expect_folder "$HOME/.git-cache"
```

Comment:
I'd avoid this test... unless you're running the test in a sandbox with dedicated HOME dir... it's intrusive to the developer's HOME dir.

Reply:
Testing suites run within a container, locally and CI.